### PR TITLE
Add more vector methods

### LINKED ISSUE #116: [Rust] Implement more methods in `Vector`
Planus vectors are basically just slices with extra steps. They should support more operations like splitting, trimming or more iterators (windows/chunks). Basically just look at everything from [the std slice](https://doc.rust-lang.org/std/primitive.slice.html) and see if it makes sense to port it.

When implementing this, keep in mind that most of the rust standard library was made before const generics -- some of the functions would look a bit different if they were designed today.

### PR DESCRIPTION
**Pull request by TethysSvensson**
Fixes #116.

**Checklist**
- [x] Updated CHANGELOG.md with relevant changes
- [x] Added tests for any new/fixed functionality
- [x] Added/updated documentation for new/changed code
- [x] Checked that README.md still makes sense (and updated it if necessary)

### CODE REVIEW

**Comment by TethysSvensson** [comment]
The only real open question here is whether we should implement `Index` which panics, like the `std` slice does.

**Comment by TethysSvensson** [comment]
> After talking to @nicklaswj I've decided that we do in fact want to implement the `Index` trait.

And after trying to do this, I have realized that the `Index` trait requires the output to be an `&T`, which we cannot give.

File: planus/src/vectors/vector.rs [hunk 1]
```diff
@@ -0,0 +1,580 @@
...
+    #[must_use]
+    pub fn split_at(self, mid: usize) -> Option<(Vector<'buf, T>, Vector<'buf, T>)> {
+        if mid <= self.len {
+            Some(unsafe { self.split_at_unchecked(mid) })
+        } else {
+            None
+        }
+    }
+
+    /// Divides one `Vector`] into two at an index, without doing bounds checking.
```
Comment (kristoff3r) [inline planus/src/vectors/vector.rs:271]:
```suggestion
    /// Divides one [`Vector`] into two at an index, without doing bounds checking.
```

File: test/rust/api_files/vector_iteration.rs [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+#[track_caller]
+fn eq<I1, I2>(iter1: I1, iter2: I2)
+where
+    I1: Clone + Iterator + ExactSizeIterator,
+    I2: Clone + Iterator + ExactSizeIterator,
+    I1::Item: std::fmt::Debug + PartialEq<I2::Item>,
+    I2::Item: std::fmt::Debug,
+{
+    // println!("iter1={iter1:?}");
```
Comment (kristoff3r) [inline test/rust/api_files/vector_iteration.rs:101]: dead code

**Review by kristoff3r — APPROVED** [review]

**Pull request merged**